src/platform/unix/mod.rs

```diff
@@ -271,15 +271,14 @@ impl OsIpcSender {
                     },
                 ];
 
-                let msghdr = msghdr {
-                    msg_name: ptr::null_mut(),
-                    msg_namelen: 0,
-                    msg_iov: iovec.as_mut_ptr(),
-                    msg_iovlen: iovec.len() as IovLen,
-                    msg_control: cmsg_buffer as *mut c_void,
-                    msg_controllen: cmsg_space as MsgControlLen,
-                    msg_flags: 0,
-                };
+                let mut msghdr: msghdr = unsafe { mem::zeroed() };
+                msghdr.msg_name = ptr::null_mut();
+                msghdr.msg_namelen = 0;
+                msghdr.msg_iov = iovec.as_mut_ptr();
+                msghdr.msg_iovlen = iovec.len() as IovLen;
+                msghdr.msg_control = cmsg_buffer as *mut c_void;
+                msghdr.msg_controllen = cmsg_space as MsgControlLen;
+                msghdr.msg_flags = 0;
```

Comment:
Could you wrap this into a function? E.g. `new_msghdr`?

Reply:
Sure, re-uploaded.

Reply:
Sorry for the noise, I seem to misunderstand how to extract this piece. Will get back to this.